### PR TITLE
Cyborg Monitoring Improvements

### PR DESCRIPTION
**Pull request by itsmeow**
## About The Pull Request

Ports some improvements to Cyborg Monitoring app, and adds a syndicate version for nukies.

- https://github.com/tgstation/tgstation/pull/50691
- https://github.com/tgstation/tgstation/pull/55593

Also includes logging and an IC filter check for the app.

## Why It's Good For The Game

Nukies can track syndieborgs and message them, the monitor app is generally better, has more feedback, and more interactions with the game. Now with 100% gamer filter.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

image

image

image

image

image

image

</details>

## Changelog
:cl:
qol: The Cyborg Remote Monitoring app has auditory and text feedback when sending a message.
tweak: Emagged tablets can use Cyborg Monitoring's message feature without an ID inserted.
add: Added a syndicate version of Cyborg Monitoring for messaging/monitoring syndicate borgs.
admin: Added an IC filter check and logging to Cyborg Monitoring messages.
/:cl:

### CODE REVIEW

**Comment by Coding-Grape** [comment]
I don't think that this should have an IC filter if PDA messaging and the tablet chat rooms don't have an IC filter

**Comment by itsmeow** [comment]
PDA messages and chatrooms both have IC filters.

PDA has always had it, I added it to chatrooms not long ago as requested by the admins.

They're having internal discussion about allowing limited netspeak on these forms of communication, but for now they should be treated like PDAs.

**Comment by Coding-Grape** [comment]
> PDA messages and chatrooms both have IC filters.
> 
> PDA has always had it, I added it to chatrooms not long ago as requested by the admins.
> 
> They're having internal discussion about allowing limited netspeak on these forms of communication, but for now they should be treated like PDAs.

That's... really fucking dumb? You're physically typing these in an informal form of communication in all 3 of these situations 
I'm not saying let people say sussy imposter and shit, just let them type IDK and LOL and such other shorthands

**Comment by itsmeow** [comment]
The assumption is that will be allowed however it requires adding a new filter type - the filter config is all serverside and handled by admin so you can't really add to it with a PR without working with server ops team.

Anyway, discussions like this can be done in Discord or forums. No need to crowd the PR with unrelated comments.

**Review by Penwin0 — DISMISSED** [review]

File: code/modules/modular_computers/file_system/programs/borg_monitor.dm [hunk 1]
```diff
@@ -58,17 +62,61 @@
...
+	available_on_ntnet = FALSE
+	available_on_syndinet = TRUE
+	transfer_access = null
+	tgui_id = "NtosCyborgRemoteMonitorSyndicate"
+
+/datum/computer_file/program/borg_monitor/syndicate/run_emag()
+	return FALSE
+
+/datum/computer_file/program/borg_monitor/syndicate/evaluate_borg(mob/living/silicon/robot/R)
+	if((get_turf(computer)).get_virtual_z_level() != (get_turf(R)).get_virtual_z_level())
```
Comment (PowerfulBacon) [inline code/modules/modular_computers/file_system/programs/borg_monitor.dm:120]: This compiles??
This just makes my byond sus senses go off.

Comment (PowerfulBacon) [reply code/modules/modular_computers/file_system/programs/borg_monitor.dm:120]: I would feel much more at peace if these were explicitly typed coming out of, and then those variables were acted on. (var/turf/computer_turf = get_turf(computer))

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**